### DISCUSSION TITLE
Set the (persisted data | storage Item) manually

### DISCUSSION THREAD

**Discussion by fitimbytyqi** [pmndrs/zustand#1448] [Q&A] [answered]
Hello community, 

First of all, thanks for creating this absolutely amazing library that I enjoy using.

I'm dealing with a problem where in my application, I save some user input like email & verification code to be used in other screens.

I'm using persist middleware to persist the data in sessionStorage but I've noticed that every time the application runs the initial state gets stored in the sessionStorage because I've called the store somewhere in any component and I don't want that to happen.

I'd rather want the data to be stored in sessionStorage when a button is clicked and be removed in the same way (manually remove key&value pair from storage and stop it from being added automatically on the next refresh of the application).

My current store is like so.

```mport create from 'zustand';
import { persist } from 'zustand/middleware';

type Nullable<T> = T | null;
type DataType = {
  email: string;
  code: string;
};

interface IAccountRecovery {
  accountRecoveryData: Nullable<DataType>;
  setAccountRecoveryData: (accountRecoveryData: Nullable<DataType>) => void;
  clearAccountRecoveryData: () => void;
}

export const useAccountRecoveryStore = create<
  IAccountRecovery,
  [['zustand/persist', IAccountRecovery]]
>(
  persist(
    (set) => ({
      accountRecoveryData: null,
      setAccountRecoveryData: (accountRecoveryData: Nullable<DataType>) =>
        set(() => ({ accountRecoveryData })),
      clearAccountRecoveryData: () =>
        set(() => ({ accountRecoveryData: null })),
    }),
    {
      name: 'account.r',
      getStorage: () => sessionStorage,
    },
  ),
);
```

I've tried clearing the store using clearStorage when I don't need the store data anymore but in the next refresh the key&value pair gets added automatically in the sessionStorage and I don't really want it to be there always.

Am I missing anything?

Thanks for your time :)

**Comment by dai-shi** [comment, marked as answer]
Sounds like you don't want to use `persist` and instead save to / load from sessionStorage manually on your end?

**Reply by fitimbytyqi** [reply]
Yes, that sounds like it but I wanted to achieve that using zustand but seems like thats not possible and just thinking about it I could use storage methods to achieve it.

Thanks a lot,
